### ISSUE TITLE
CF - errors are occurring occasionally when retrieving the space from an application

### ISSUE THREAD

**Issue by Jammy-Louie** [spinnaker/spinnaker#3977]
### Issue Summary:
There are reports that an unknown error is being thrown from the cf clouddriver.

```
2019-02-13 08:49:24.057  WARN 1 --- [utionAction-190] c.n.s.c.cache.LoggingInstrumentation     : com.netflix.spinnaker.clouddriver.cloudfoundry.provider.CloudFoundryProvider:dev.na-4a.gaia/CloudFoundryCachingAgent completed with one or more failures

com.netflix.spinnaker.clouddriver.cloudfoundry.client.CloudFoundryApiException: Cloud Foundry API returned with error(s): An unknown error occurred.

                at com.netflix.spinnaker.clouddriver.cloudfoundry.client.CloudFoundryClientUtils.safelyCall(CloudFoundryClientUtils.java:48) ~[clouddriver-cloudfoundry.jar:na]

                at com.netflix.spinnaker.clouddriver.cloudfoundry.client.Applications.map(Applications.java:102) ~[clouddriver-cloudfoundry.jar:na]

                at java.util.Optional.map(Optional.java:215) ~[na:1.8.0_181]

                at com.netflix.spinnaker.clouddriver.cloudfoundry.client.Applications.findById(Applications.java:59) ~[clouddriver-cloudfoundry.jar:na]

                at com.netflix.spinnaker.clouddriver.cloudfoundry.client.Routes.lambda$map$0(Routes.java:74) ~[clouddriver-cloudfoundry.jar:na]

                at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193) ~[na:1.8.0_181]

                at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1382) ~[na:1.8.0_181]

                at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481) ~[na:1.8.0_181]

                at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471) ~[na:1.8.0_181]

                at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:708) ~[na:1.8.0_181]

                at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234) ~[na:1.8.0_181]

                at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:499) ~[na:1.8.0_181]

                at com.netflix.spinnaker.clouddriver.cloudfoundry.client.Routes.map(Routes.java:75) ~[clouddriver-cloudfoundry.jar:na]

                at com.netflix.spinnaker.clouddriver.cloudfoundry.client.Routes.all(Routes.java:137) ~[clouddriver-cloudfoundry.jar:na]

                at com.netflix.spinnaker.clouddriver.cloudfoundry.provider.agent.CloudFoundryCachingAgent.loadData(CloudFoundryCachingAgent.java:74) ~[clouddriver-cloudfoundry.jar:na]

                at com.netflix.spinnaker.cats.agent.CachingAgent$CacheExecution.executeAgentWithoutStore(CachingAgent.java:80) ~[cats-core.jar:na]

                at com.netflix.spinnaker.cats.agent.CachingAgent$CacheExecution.executeAgent(CachingAgent.java:73) ~[cats-core.jar:na]

                at com.netflix.spinnaker.cats.redis.cluster.ClusteredAgentScheduler$AgentExecutionAction.execute(ClusteredAgentScheduler.java:299) ~[cats-redis.jar:na]

                at com.netflix.spinnaker.cats.redis.cluster.ClusteredAgentScheduler$AgentJob.run(ClusteredAgentScheduler.java:273) ~[cats-redis.jar:na]

                at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) ~[na:1.8.0_181]

                at java.util.concurrent.FutureTask.run(FutureTask.java:266) ~[na:1.8.0_181]

                at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [na:1.8.0_181]

                at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [na:1.8.0_181]

                at java.lang.Thread.run(Thread.java:748) [na:1.8.0_181]
```

### Cloud Provider(s):
CF

### Feature Area: 
Clouddriver

### Description:
We should investigate what conditions would cause an error when retrieving a space from an application. We should also handle and surface better error message.

**Comment by Jammy-Louie** [comment]
@spinnakerbot add-label provider/cf

**Comment by jkschneider** [comment]
Pas Apicella saw a failure on the same line for a certain set of credentials in PWS.

**Comment by jmelchio** [comment]
@jkschneider do we know what the permissions on the credentials were for a given org and space?

**Comment by jmelchio** [comment]
@Jammy-Louie do we have a contact we can talk to so we can get some information on how to re-create this problem? Until we can do that it is hard to move ahead with this one. 

Experiments with accounts have shown some interesting things but we were not able to re-create the issue.

**Comment by Jammy-Louie** [comment]
it looks like this is a possible out of memory issue as reported here https://github.com/cloudfoundry/cloud_controller_ng/issues/864

the release for the reported issue was fixed in `CAPI release 1.44.0` this corresponds to PCF 2.0 releases
